pygotham/frontend/__init__.py

```diff
@@ -60,9 +63,29 @@ def create_app(settings_override=None):
     assets.register('css_foundation', css_foundation)
     assets.register('js_foundation', js_foundation)
 
+    @app.url_defaults
+    def add_event_slug(endpoint, values):
+        if 'event_slug' in values or not g.current_event:
+            return
+        if app.url_map.is_endpoint_expecting(endpoint, 'event_slug'):
+            values['event_slug'] = g.current_event.slug
+
+    @app.url_value_preprocessor
+    def current_event_from_url(endpoint, values):
+        if values is None:
+            values = {}
+        if app.url_map.is_endpoint_expecting(endpoint, 'event_slug'):
+            now = arrow.utcnow().to('America/New_York').naive
+            g.current_event = Event.query.filter(
+                Event.slug == values.pop('event_slug', None),
+                Event.active == True,
+                or_(Event.activity_begins == None, Event.activity_begins <= now),
+                or_(Event.activity_ends == None, Event.activity_ends > now),
```

Comment:
We can leave them. `activity_begins` is important. `activity_ends` is less important now, but could be used to turn an event off at a certain point in time.

That said, since this check runs all the time, it might be worth removing `activity_ends`.